app/src/main/java/com/sdp/movemeet/map/MainMapFragment.java

```diff
@@ -0,0 +1,208 @@
+            //if (user == null) intent = new Intent(MapsActivity.this, ActivityDescriptionActivityUnregistered.class);
+            //else
+            intent = new Intent(supportMapFragment.getActivity(), ActivityDescriptionActivity.class);
+            intent.putExtra("activity", act);
+            startActivity(intent);
+        }
+        return true;
+    }
+
+    @Override
+    public void onMapClick(LatLng latLng) {
+        Log.d(TAG, "map is clicked");
+        if (newActivityMarker != null) newActivityMarker.remove();
+        //Only if user is logged we allow the possibility of creating a new activity
+        if (user != null) {
+            Log.d(TAG, "user is logged");
+            Marker newActivity = googleMap.addMarker(
+                    new MarkerOptions().position(latLng).title("new activity").snippet("Click her to create new activity")
+            );
+            newActivityMarker = newActivity;
+            newActivityMarker.setTag("new activity position");
+            newActivity.showInfoWindow();
+        }
+    }
+
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private void getNearbyMarkers(GoogleMap googleMap) {
+        DistanceCalculator dc = new DistanceCalculator(currentLocation.getLatitude(), currentLocation.getLongitude());
+
+        dc.setActivities((ArrayList<Activity>) activities);
+        dc.calculateDistances();
+        dc.sort();
+
+        for (Activity act : dc.getTopActivities(activities.size())) {
+            LatLng actLatLng = new LatLng(act.getLatitude(), act.getLongitude());
+
+            MarkerOptions markerOpt = new MarkerOptions().position(actLatLng).title(act.getTitle());
+            markerOpt.icon(BitmapDescriptorFactory.fromResource(chooseIcon(act)));
+            Marker marker = googleMap.addMarker(markerOpt);
+            marker.setTag(act);
+        }
+    }
+
+
+
+    private int chooseIcon(Activity activity) {
+        switch (activity.getSport()) {
+            case Soccer:
+                return R.drawable.icon_soccer;
+            case Running:
+                return R.drawable.icon_running;
+            case Tennis:
+                return R.drawable.icon_tennis;
+            case Swimming:
+                return R.drawable.icon_swim;
+            case Badminton:
+                return R.drawable.icon_badminton;
+            default:
+                return -1;
+        }
+    }
+
+    public GoogleMap getGoogleMap() {
+        return googleMap;
+    }
+
+
+    @Override
+    public void onInfoWindowClick(Marker marker) {
+        newActivityMarker.remove();
+        Intent intent = new Intent(supportMapFragment.getActivity(), UploadActivityActivity.class);
+        //use Bundle to pass latlng instance to intent
+        Bundle arg = new Bundle();
+        arg.putParcelable("position",marker.getPosition());
+        intent.putExtra("bundle", arg);
+        //remove marker
+        startActivity(intent);
+    }
+}
```

Comment:
Very good job by the way! It's a clear file and we can easily understand the main lines by going through it even if we don't work on the map features! Bravo!

Reply:
thank you !